server/src/main/java/org/folio/metastorage/server/MetaStorageService.java

```diff
@@ -53,6 +53,23 @@ Future<Void> putGlobalRecords(RoutingContext ctx) {
     }
   }
 
+  Future<Void> reloadCodeModule(RoutingContext ctx) {
+    RequestParameters params = ctx.get(ValidationHandler.REQUEST_CONTEXT_KEY);
+    String id = Util.getParameterString(params.pathParameter("id"));
+    Storage storage = new Storage(ctx);
+    return storage.selectCodeModuleEntity(id)
+        .onSuccess(e -> {
+          if (e == null) {
+            HttpResponse.responseError(ctx, 404,
+                String.format(ENTITY_ID_NOT_FOUND_PATTERN, MODULE_LABEL, id));
+            return;
+          }
+          ModuleCache.getInstance().purge(TenantUtil.tenant(ctx), id);
```

Comment:
@adamdickmeiss any reason why we don't do "lookup" here as well to re-populate the cache?

Reply:
Cache is populated anyway, but doing lookup here will ensure "new" failures are caught early.